### PR TITLE
Correct misclassification of .js.erb files

### PR DESCRIPTION
**Pull request by CHTJonas**
Change the language classification of `*.js.erb` files from HTML+ERB to JavaScript+ERB. I think I've covered all the relevant bases but let me know if there's something I can correct - thanks for taking the time to review! 😄 

## Description
Previously JavaScript files containing embedded Ruby syntax were misclassified as HTML+ERB. This PR would correct this by classifying such source code files as JavaScript+ERB and grouping under the parent JavaScript language.

Motivation for this is that JavaScript files meant for client-side execution in response to an AJAX request in a Rails application are listed under HTML when running `github-linguist --breakdown`.

## Checklist:
- [x] **I am associating a language with a new file extension.**
  - [x] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      - https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3Aerb+escape_javascript+NOT+nothack
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://edgeguides.rubyonrails.org/working_with_javascript_in_rails.html#a-simple-example
    - Sample license(s):
      - MIT
  - [x] I have included a change to the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -2371,6 +2371,16 @@ JavaScript:
   interpreters:
   - node
   language_id: 183
+JavaScript+ERB:
+  type: programming
+  tm_scope: none
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: Couldn't we use the JavaScript grammar for highlighting here?

File: lib/linguist/languages.yml [hunk 2]
```diff
@@ -2371,6 +2371,16 @@ JavaScript:
...
+  group: JavaScript
+  extensions:
+  - ".js.erb"
+  ace_mode: text
+  codemirror_mode: htmlembedded
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: The [HTML Embedded](https://codemirror.net/mode/htmlembedded/index.html) page mentions that it supports the `x-erb` MIME type, but did you check how the actual highlighting looks like? Wouldn't JavaScript be a better fit here as well?

**Comment by CHTJonas** [comment]
Thanks for the comments @pchaigno! Syntax highlighting updated now to use JavaScript now.

I wasn't sure whether `tm_scope` should be `source.js` (same as JavaScript language proper) or whether I should add `text.js.erb` to the `grammars.yml` file to match what HTML+ERB does?

**Comment by pchaigno** [comment]
> I wasn't sure whether `tm_scope` should be `source.js` (same as JavaScript language proper) or whether I should add something like `text.js.erb` to the `grammars.yml` file to match what HTML+ERB and NetLinx+ERB do?

The `grammar.yml` should not be manually updated so `source.js` is fine. If we're using JavaScript for both the TextMate scope and the CodeMirror grammar, we might as well add the extension to the list of JavaScript extensions. It would achieve the exact same effect.

**Comment by CHTJonas** [comment]
I was considering that, but I think(?) listing separately does allow some benefits eg. when searching you can restrict the scope to JavaScript+ERB as opposed to all JavaScript, such as [this example](https://github.com/RailsApps/rails-composer/search?l=HTML%2BERB) with HTML+ERB.

**Comment by CHTJonas** [comment]
Did you have any thoughts on the above RE: whether to keep the `.js.erb` extension separate or just add it to the list of JS extensions @pchaigno?

**Pull request merged**